x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/rolemapping/TransportGetRoleMappingsAction.java

```diff
@@ -56,14 +57,21 @@ public TransportGetRoleMappingsAction(
     protected void doExecute(Task task, final GetRoleMappingsRequest request, final ActionListener<GetRoleMappingsResponse> listener) {
         final Set<String> names;
         if (request.getNames() == null || request.getNames().length == 0) {
-            names = null;
+            names = Set.of();
         } else {
             names = new HashSet<>(Arrays.asList(request.getNames()));
         }
         roleMappingStore.getRoleMappings(names, ActionListener.wrap(mappings -> {
             List<ExpressionRoleMapping> combinedRoleMappings = Stream.concat(
                 mappings.stream(),
-                clusterStateRoleMapper.getMappings(names)
+                clusterStateRoleMapper.getMappings(names.stream().map(name -> {
+                    // If a read-only role is fetched by name including suffix, remove suffix
+                    if (name.length() >= RESERVED_ROLE_MAPPING_SUFFIX.length()) {
```

Comment:
I think we want an endsWith check here, otherwise a long enough name will get borked